django/test/testcases.py

```diff
@@ -756,6 +756,36 @@ def assertXMLNotEqual(self, xml1, xml2, msg=None):
                 standardMsg = '%s == %s' % (safe_repr(xml1, True), safe_repr(xml2, True))
                 self.fail(self._formatMessage(msg, standardMsg))
 
+    def assertFormValid(self, form, msg=None):
+        """
+        Asserts that a form is valid
+        """
+        if not form.is_valid():
+            standardMsg = "Form validation failed. The following field(s) had errors:\n"
+            for field, field_errors in form.errors.items():
+                standardMsg += "- %s:\n" % field
+                for error in field_errors:
+                    standardMsg += "    - %s\n" % error
+            self.fail(self._formatMessage(msg, standardMsg))
+
+    def assertFormNotValid(self, form, expected_errors=None, msg=None):
```

Comment:
By the way, I don't find `assertFormNotValid` compelling as we already have `assertFormError`.